psi4/src/psi4/libfock/PKmanagers.cc

```diff
@@ -107,24 +110,48 @@ std::shared_ptr<PKManager> PKManager::build_PKManager(std::shared_ptr<PSIO> psio
         ncorebuf = 3;
     }
 
+    // determine which sub-algorithm to use
     bool do_reord = false;
     bool do_yosh = false;
     bool do_incore = false;
-    if (options["PK_ALGO"].has_changed()) {
-        if (algo == "REORDER") {
-            do_reord = true;
-        } else if (algo == "YOSHIMINE") {
-            do_yosh = true;
-        }
-    } else {
+
+    // specify particular out-of-core subalgorithm...
+    if (subalgo == "OOC_REORDER") {
+	do_reord = true;
+    } else if (subalgo == "OOC_YOSHIMINE") {
+        do_yosh = true;
+
+    // ...or automatically select an out-of-core subalgorithm...
+    } else if (subalgo == "OUT_OF_CORE") {
         if (algo_factor * memory > pk_size) {
             do_reord = true;
         } else {
             do_yosh = true;
         }
-    }
 
-    if (ncorebuf * pk_size < memory && !noincore) do_incore = true;
+    // ...or force the in-core algorithm...
+    } else if (subalgo == "INCORE") {
+	// throw an exception if in-core is forced, but not enough memory is allocated
+        if (ncorebuf * pk_size > memory) {
+            throw PSIEXCEPTION("SCF_SUBTYPE=INCORE was specified, but there is not enough memory to do in-core! Increase the amount of memory allocated to Psi4 or allow for out-of-core to be used.\n");
+        } else {
+            do_incore = true;
+	}
+
+    // ...or just let psi4 pick any subalgorithm
+    } else if (subalgo == "AUTO") {
+        if (ncorebuf * pk_size < memory) {
+            do_incore = true;
+	} else if (algo_factor * memory > pk_size) {
+            do_reord = true;
+	} else {
+            do_yosh = true;
+	}
+
+    // throw an exception on an invalid SCF_SUBTYPE
+    } else {
+        throw PSIEXCEPTION("Invalid SCF_SUBTYPE option! The valid choices of SCF_SUBTYPE for SCF_TYPE=PK are AUTO, INCORE, OUT_OF_CORE, OOC_YOSHIMINE, and OOC_REORDER.");
+    }
```

Comment:
`OOC_YOSHIMINE` and `OOC_REORDER` need switching out in this file, too.

Reply:
Oh right, forgot to switch that out. Will do!

Reply:
Done and done!